stor/stor/utils.py

```diff
@@ -410,19 +279,13 @@ def copy(source, dest, **kwargs):
             ...
             ValueError: OBS destination must be file with extension or directory with slash
     """
-    from stor import Path
     from stor.obs import OBSUploadObject
-
-    source = Path(source)
-    dest = Path(dest)
+    source = stor.Path(source)
+    dest = stor.Path(dest)
```

Comment:
It seems like this would become a lot simpler if we made these into shared methods rather than into utils.

for the POSIX implementation, you always know that source is a file or directory, so the other checks are not required.

for the OBS implementation, for Swift and S3, non-POSIX dest is not supported; for DX, only other DX is supported. (I'm thinking you could write one method along with a `supports_server_side_copy` flag that's True for DX and False for Swift and S3)

Reply:
Done as part of https://github.com/counsyl/stor/pull/98